src/components/Header.tsx

```diff
@@ -16,7 +16,8 @@ export const HEADER_HEIGHT = Platform.select({
   android: 38,
 }) as number;
 
-interface Props extends Partial<NavigationScreenProps> {
+interface Props {
+  navigation?: StackNavigationProp<any, any> | any;
```

Comment:
about TypeScript, I know it's still WIP but consider using alias type for `navigation` somewhere in your code
like:
```
type NavigationProp = StackNavigationProp<any, any> | any;
```
and use it in your components. 
Having this additional abstraction layer should help with refactoring in the future.